index.bs

```diff
@@ -159,11 +159,29 @@ Note: These steps have to be run on the [=file system queue=].
 </div>
 
 <div algorithm>
-To <dfn for="file entry/lock">release</dfn> a [=file entry/lock=] on a given
-[=file entry=] |file|:
+To <dfn for="file system entry">check for a descendant lock</dfn> on a given
+[=directory entry=] |directory|:
+
+1. [=set/For each=] |child| of |directory|'s [=directory entry/children=]:
+  1. Let |lock| be the |child|'s [=file system entry/lock=].
+  1. If |lock| is not "`open`":
+    1. Return "`taken`".
+  1. If |child| is a [=directory entry=]:
+    1. Let |descendantLockStatus| be the result of [=file system entry/checking for a descendant lock=] on |child|.
+    1. If |descendantLockStatus| is "`taken`":
+      1. Return "`taken`".
+1. Return "`open`".
 
-1. Let |lock| be the |file|'s associated [=file entry/lock=].
-1. Let |count| be the |file|'s [=file entry/shared lock count=].
+Note: These steps have to be run on the [=file system queue=].
```

Comment:
This makes more sense to note at the bottom of "take a lock" as that's the one that will be invoked directly from elsewhere, no?

Reply:
This is noted at the bottom of "take a lock". Should it only be noted there?